manifests/manage_binding.pp

```diff
@@ -33,22 +33,9 @@
         fail('https bindings require a valid ip_address')
       }
 
-      file { "inspect-${title}-certificate.ps1":
-        ensure  => file,
-        path    => "C:\\temp\\inspect-${name}.ps1",
-        content => template('iis/inspect-certificate-binding.ps1.erb'),
-      }
-
-      file { "create-${title}-certificate.ps1":
-        ensure  => file,
-        path    => "C:\\temp\\create-${name}.ps1",
-        content => template('iis/create-certificate-binding.ps1.erb'),
-      }
-
       exec { "Attach-Certificate-${title}":
-        command   => "C:\\temp\\create-${name}.ps1",
-        onlyif    => "C:\\temp\\inspect-${name}.ps1",
-        require   => [File["inspect-${title}-certificate.ps1"], File["create-${title}-certificate.ps1"]],
+        command   => template('iis/create-certificate-binding.ps1.erb'),
```

Comment:
It is documented.
"Alternately, you can put the PowerShell code for the command, onlyif, and unless parameters into separate files, and then invoke the file function in the resource. You could also use templates and the template() function if the PowerShell scripts need access to variables from Puppet."

https://forge.puppet.com/puppetlabs/powershell/readme#beginning-with-powershell

My understanding is the no file is created.  The command is rendered on the server and sent inline to the client.   This makes variables passing simpler.